src/traversal/writeRelayQueryPayload.js

```diff
@@ -33,8 +34,15 @@ function writeRelayQueryPayload(
   var store = writer.getRecordStore();
   var path = new RelayQueryPath(query);
 
+
   RelayNodeInterface.getResultsFromPayload(store, query, payload)
     .forEach(({dataID, result}) => {
+      invariant(
+        typeof dataID === 'string',
+        'writeRelayQueryPayload: expected ID on the payload for query %s to be string' +
+        ', instead received the %s %s',
+        path.getName(), typeof dataID, dataID
```

Comment:
A few nits on formatting:
- all args on their own line
- in the template string, wrap `%s` with backticks so `` `%s` ``
- ending period on the sentence